### PR TITLE
Object error handling implemented

### PR DESCRIPTION
**Pull request by Holado**
Dependent on changes in pull request for branch argsrefine in GreynirPackage

* Error information from Verbs.conf for object errors used in error handling

### CODE REVIEW

**Comment by Holado** [comment]
Pytest was fixed; the OBJ_ERRORS mypy complaint is due to changes in GreynirPackage in argsrefine.

File: src/reynir_correct/errfinder.py [hunk 1]
```diff
@@ -697,17 +700,46 @@ def find_verb_subject(tnode: SimpleTree) -> Optional[SimpleTree]:
...
                     subj = p.NP_SUBJ
                 except AttributeError:
                     pass
         return subj
 
+    @staticmethod
+    def find_verb_direct_object(tnode: SimpleTree) -> Optional[SimpleTree]:
+        """ Starting with a verb terminal node, attempt to find the 
+        verb's direct object noun phrase """
+        obj = None
```
Comment (vthorsteinsson) [inline src/reynir_correct/errfinder.py:718]: I would have thought you needed a type annotation here: `obj: Optional[SimpleTree] = None`

**Review by vthorsteinsson — COMMENTED** [review]
Apart from a single type annotation nitpick, this looks good to me!

File: src/reynir_correct/errfinder.py (hunk 1 above)
Comment (Holado) [reply src/reynir_correct/errfinder.py:718]: Copied this mostly from find_verb_subject(); fixed in both places

**Comment by Holado** [comment]
Review issues have been fixed; this should be merged after corresponding GreynirPackage pull request has been merged, as it breaks the system otherwise.

**Comment by Holado** [comment]
This should be ready for merging at the same time as argsrefine in GreynirPackage

File: src/reynir_correct/errfinder.py [hunk 2]
```diff
@@ -207,6 +208,12 @@ def verb_subject_matches(cls, verb: str, subj: str) -> bool:
             verb, set()
         ) or subj in cls._VERB_ERROR_SUBJECTS.get(verb, dict())
 
+    @classmethod
+    def verb_matches_arguments(cls, key: str) -> bool:
+        """ Return True if the given arguments are allowed for this verb 
+        or if these are erroneous arguments which we can flag """
+        # This is overridden in reynir_correct.checker
```
Comment (vthorsteinsson) [inline src/reynir_correct/errfinder.py:215]: Is this comment correct (also in GreynirPackage), since this is actually errfinder.py?

Comment (vthorsteinsson) [reply src/reynir_correct/errfinder.py:215]: ...and how is an erroneous verb frame then detected and reported to the user as an annotation?

Comment (Holado) [reply src/reynir_correct/errfinder.py:215]: Copied from similar files in GreynirPackage, fixed.
This ensures erroneous verb frames are allowed in the parsing process. Without this, sentences such as 'Ég rústa bílnum' are not parsable. 
The error is then detected in check_obj() under _annotate_verb() in errfinder.py, and then annotated with a call to annotate_wrong_obj_case().

**Pull request merged**